src/drive/web/modules/navigation/duck/reducer.js

```diff
@@ -188,45 +256,112 @@ const indexFor = (file, array, compareFn, start = 0, end = array.length) => {
 }
 
 // reducer for the full file list of the currently displayed folder
-const files = (state = [], action) => {
+const files = (
+  state = { shared: [], recent: [], trashed: [], folder: [] },
+  action
+) => {
   switch (action.type) {
+    case OPEN_FOLDER_FROM_SHARINGS_SUCCESS:
+      return {
+        ...state,
+        shared: action.files
+      }
+    case OPEN_FOLDER_FROM_TRASH_SUCCESS:
+      return {
+        ...state,
+        trashed: action.files
+      }
     case OPEN_FOLDER_SUCCESS:
+      /* 
+        Special case for the Trash dir since `Trash` has its own 
+        reducer to store its files and that Trash is a "regular"
+        folder
+        */
+      if (action.folder.id === TRASH_DIR_ID) {
+        return {
+          ...state,
+          trashed: action.files
+        }
+      }
```

Comment:
I think we can remove this special case since now we use `OPEN_FOLDER_FROM_TRASH` right?

Reply:
Yep.